modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -428,7 +429,7 @@ void executeSingleUpload(
         final PutObjectRequest putRequest = new PutObjectRequest(s3BlobStore.bucket(), blobName, input, md);
         putRequest.setStorageClass(s3BlobStore.getStorageClass());
         putRequest.setCannedAcl(s3BlobStore.getCannedACL());
-        putRequest.setRequestMetricCollector(s3BlobStore.putMetricCollector);
+        putRequest.setRequestMetricCollector(blobStore.getMetricCollector(Operation.PUT_OBJECT, purpose));
```

Comment:
The rest of this method uses the `s3BlobStore` argument rather than the `blobStore` field. I'm not sure why, maybe we can clean that up, but for now let's not change that.

```suggestion
        putRequest.setRequestMetricCollector(s3BlobStore.getMetricCollector(Operation.PUT_OBJECT, purpose));
```